### PR TITLE
adding export copy functionality

### PR DESCRIPTION
**Pull request by calellowitz**
@NoahCarnahan cc: @sheelio 
Copy export functionality
screenshot:
screen shot 2017-03-06 at 3 51 35 pm

### CODE REVIEW

File: corehq/apps/reports/static/reports/js/export.manager.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ var ExportManager = function (o) {
     self.bulkDownloadUrl = o.bulkDownloadUrl;
     self.exportFilters = o.exportFilters;
     self.jsonExportFilters = o.jsonExportFilters;
+    self.copyUrl = o.copyUrl;
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/js/export.manager.js]: Expected indentation of 8 space characters but found 4. (indent)

**Review by NoahCarnahan — APPROVED** [review]
Looks good thanks Cal

File: corehq/apps/export/models/new.py [hunk 2]
```diff
@@ -844,6 +844,13 @@ def get_payload(self, stream=False):
         """
         return self.fetch_attachment(DAILY_SAVED_EXPORT_ATTACHMENT_NAME, stream=stream)
 
+    def copy_export(self):
+        export_json = self.to_json()
+        del export_json['_id']
+        export_json['name'] = '{} - Copy'.format(self.name)
+        new_export = self.__class__.wrap(export_json)
+        new_export.save()
```
Comment (benrudolph) [inline corehq/apps/export/models/new.py]: nit: our architecture guide prefers returning the object and allowing for the client to save the model: https://github.com/dimagi/architecture-guide/blob/master/let_client_save_model.md#best-practice-let-the-client-code-save-the-model

File: corehq/apps/export/views.py [hunk 3]
```diff
@@ -2415,3 +2418,23 @@ def download_daily_saved_export(req, domain, export_instance_id):
...
+    def dispatch(self, request, *args, **kwargs):
+        if not self.request.couch_user.can_edit_data():
+            raise Http404
+        else:
+            return super(CopyExportView, self).dispatch(request, *args, **kwargs)
+
+    def get(self, request, domain, export_id, *args, **kwargs):
+        try:
+            export = get_properly_wrapped_export_instance(export_id)
+            export.copy_export()
```
Comment (benrudolph) [inline corehq/apps/export/views.py]: nit this could go in the `else` block of the try statement since it won't throw a ResourceNotFound error

**Review by benrudolph — COMMENTED** [review]
nice looks good. do you think we could add a test for this? also this is more of a product question, but do we want to make this available to everyone or have it behind a feature flag? i'm just worried this will encourage people making a _lot_ of daily saved exports which are notoriously tough on our servers

**Comment by sheelio** [comment]
I think that it would be OK to release for everyone (since we moved daily
saved exports to a paid plan and we stop refreshing the daily saved exports
if they are not in use).

**Comment by calellowitz** [comment]
@benrudolph updated

**Comment by benrudolph** [comment]
Thanks! Looks good

**Pull request merged**